calyx/src/passes/group_to_invoke.rs

```diff
@@ -91,6 +91,7 @@ fn construct_invoke(
 
     let mut inputs = Vec::new();
     let mut comb_assigns = Vec::new();
+    let mut wire_map: HashMap<ir::Id, ir::RRC<ir::Cell>> = HashMap::new();
```

Comment:
You could consider keeping track of the input port of the wire so that you don’t need to use .remove to gain ownership of the cell every time you use it